pynicotine/gtkgui/search.py

```diff
@@ -1709,21 +1709,24 @@ def on_filter_entry_deleted_text(self, buffer, *_args):
 
     def on_filter_entry_icon_press(self, entry, *_args):
 
-        history = config.sections["searches"].get(entry.filter_id)
-        recall_text = history[0] if history else ""
-        text = entry.get_text()
+        entry_text = entry.get_text()
+
+        if not entry_text:
+            # Recall last filter
+            history = config.sections["searches"].get(entry.filter_id)
+            recall_text = history[0] if history else ""
 
-        # Recall last filter entry if box empty
-        if not text:
-            entry.grab_focus_without_selecting()
             entry.set_text(recall_text)
             entry.set_position(-1)
+            entry.grab_focus_without_selecting()
 
-        # Activate new filter entry or Clear Filter
-        if text != recall_text:
-            self.on_refilter()
-        else:
+        elif entry_text == self.filters.get(entry.filter_id)[1]:
```

Comment:
Please unpack as `_filter_value, h_filter_value = self.filters.get(entry.filter_id)`, it's easier to understand and safer than indexing.